### PR TITLE
Cleaning up warnings and struct names

### PR DESCRIPTION
**Pull request by tyler-ball**
## Description
Cleaning up warnings by adding a bunch of comments and struct names

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tyler-ball** [comment]
@afiune I think `CookbooksStatus` is really similar to `CookbookRecord` and isn't obvious which one is which without reading the comments. We don't have to use `PipelineStatus` since we may make another reporting pipeline, but we could also move this code into a `pkg/reporting/cookbooks/` package.

What about `CbReportStatus`?

**Comment by afiune** [comment]
@tyler-ball that name it’s ok for me, I do agree both structures are similar, if you do the rename then change `CookbookRecord` as well, maybe for `CbRecord` or anything that links them. Though I don’t see the point on creating another package only for cookbooks, the reporting package is holding all reports/structs and that is ok.

Should we talk through this changes?

**Comment by afiune** [comment]
After talking in our standup we agreed that we don't want to use the word `Pipeline` in our structs or function names since that is an implementation detail that consumers don't care about.

We are going to:
* Rename `CookbooksStatus` in favor of `CookbooksReport`
* Rename `NewCookbooks()` in favor of `GenerateCookbooksReport()`
* Make all struct fields private with the exception of `Records`

**Comment by tyler-ball** [comment]
@afiune @marcparadise okay, this is ready for re-review

File: cmd/report.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ provided when the report is generated.
 			}
 
 			fmt.Println("Analyzing nodes...")
-			report, err := reporting.Nodes(chefClient.Search)
+			report, err := reporting.GenerateNodesReport(cfg, chefClient.Search)
```
Comment (afiune) [inline cmd/report.go]: `GenerateNodesReport(search)` accepts only one argument

**Comment by afiune** [comment]
Im just going to push the extra mile on this PR. 

tenor-76094038

File: pkg/reporting/cookbooks.go [hunk 2]
```diff
@@ -258,23 +264,23 @@ func (cbs *CookbooksStatus) downloadCookbook(cookbookName, version string, analy
...
+			cbr.progress.Increment()
 			return
 		}
 	}
 
 	// do we need to analyze the cookbooks
-	if cbs.RunCookstyle {
-		err = cbs.Cookbooks.DownloadTo(cookbookName, version, cbs.CookbooksDir)
+	if cbr.RunCookstyle {
+		err = cbr.cookbooks.DownloadTo(cookbookName, version, fmt.Sprintf("%s/cookbooks", analyzeCacheDir))
```
Comment (afiune) [inline pkg/reporting/cookbooks.go]: oops, this might have been a bug, we need to use `cookbooksDir` instead of this path, fixing.

**Pull request merged**